neo/core/objectlist.py

```diff
@@ -32,6 +32,10 @@ def _handle_append(self, obj):
             )
         ):
             raise TypeError(f"Object is a {type(obj)}. It should be one of {self.allowed_contents}.")
+
+        if self._contains(obj):
+            raise ValueError("Cannot add the same object twice")
```

Comment:
The string representation of data objects can be very long, so I don't think it's a good idea to put it in exception messages. How about "Cannot add this object because it is already contained within the list"?

Reply:
yeah that sounds good!

Reply:
done